docs/kratos/emails-sms/05_custom-email-templates.mdx

```diff
@@ -0,0 +1,285 @@
+
+````mdx-code-block
+<Tabs>
+  <TabItem value="cloud" label="Ory Cloud" default>
+
+You can either use in-line template definitions in nested templates. The following example defines the message body for verification emails.
+It assumes that there is a `lang` attribute defined in the identity traits that contains the preferred language of the user. The verification email
+content changes if the system detects `fr` or `it` in the identity traits. If the system detects any other language, it uses the `eng` template.
+
+```gotmpl
+{{define "en"}}
+Hi,
+
+Please recover access to your account by clicking the following link:
+
+{{ .RecoveryURL }}
+{{end}}
+
+{{define "fr"}}
+Bonjour ,
+
+Veuillez récupérer l'accès à votre compte en cliquant sur le lien suivant :
+
+{{ .RecoveryURL }}
+{{end}}
+
+{{define "it"}}
+Ciao ,
+
+Recuperare l'accesso al proprio account facendo clic sul seguente link:
+
+{{ .RecoveryURL }}
+{{end}}
+
+{{- else if eq .Identity.traits.language "fr" -}}
+{{ template "fr" . }}
+{{- else if eq .Identity.traits.language "it" -}}
+{{ template "it" . }}
+{{- else -}}
+{{ template "en" . }}
+{{- end -}}
+```
+
+  </TabItem>
+  <TabItem value="selfhosted" label="Self-Hosted Ory Kratos Config" default>
+
+To create nested templates, you can use either in-line template definitions, or as in this example, use separate templates. The following
+example defines the message body for verification emails. It assumes that there is a `lang` attribute defined in the identity traits
+that contains the preferred language of the user.
+
+```txt title="<template-root>/recovery/valid/email.body.gotmpl"
+{{- if eq .Identity.traits.language "de" -}}
+{{ template "email.body.de.gotmpl" . }}
+{{- else -}}
+{{ template "email.body.en.gotmpl" . }}
+{{- end -}}
+<a href="{{ .RecoveryURL }}">{{.RecoveryURL }}</a>
+```
+
+```txt title="<template-root>/recovery/valid/email.body.de.gotmpl"
+Hallo {{ upper .Identity.traits.firstName }},
+
+Um Ihr Konto wiederherzustellen, klicken Sie bitte auf den folgenden Link:
+```
+
+```txt title="<template-root>/recovery/valid/email.body.en.gotmpl"
+Hello {{ upper .Identity.traits.firstName }},
+
+to recover your account, please click on the link below:
+```
+
+The i18n template is nested in the root `email.body.gotmpl` template. The sub-templates must conform to the `email.body*` pattern.
+The identity of the user is available in all templates.
+
+  </TabItem>
+</Tabs>
+````
+
+:::tip
+
+You can use Sprig functions in the nested templates.
+
+:::
+
+### Nested templates containing remote templates
+
+:::warning
+
+This approach is available in self-hosted Ory Kratos instances only. For security reasons, Ory Cloud doesn't allow linking to
+external resources. As such, all options must be contained in a single template.
+
+:::
+
+When remote templates are used in self-hosted Ory Kratos, the dynamics of loading nested templates change. Remote templates can't
+reference templates outside of the instance's template directory.
```

Comment:
I don't think this makes sense. In Ory Kratos we need to use a nested template (as you have in the previous section) **when using remote templates**. So this is about using nested remote templates.

We can actually remove a lot of this section and move it to the above section with warnings. Ory Cloud **can** use remote templates and it **can** use nested templates, but when loading them in it has to be base64 encoded.

Reply:
I think I'm gonna need to jump on a call with you and learn how this nesting remote stuff mechanism works because atm I obviously don't 😅.

Reply:
sure thing :)